ci-operator/step-registry/openshift/upgrade/qe/sanity/arch-migration/openshift-upgrade-qe-sanity-arch-migration-chain.yaml

```diff
@@ -4,8 +4,9 @@ chain:
   - ref: cucushift-upgrade-prehealthcheck
   - ref: cucushift-ota-preupgrade
   - ref: cucushift-upgrade-arch-migration
-  - ref: cucushift-ota-postupgrade
   - ref: ipi-install-heterogeneous
+  - ref: cucushift-installer-check-cluster-health
+  - ref: cucushift-ota-postupgrade
```

Comment:
We could not make sure all things ready if we keep `cucushift-ota-postupgrade` just behind `cucushift-upgrade-arch-migration` for OCP 4.18 and later versions.

Reply:
I wonder why? That is why I wonder the reason of https://github.com/openshift/release/pull/58716/files#r1841794767

Reply:
As I know sometimes the MCP will not ready even cluster status reported Progressing=false, that's why we add   `cucushift-installer-check-cluster-health` before `cucushift-ota-postupgrade`.

Reply:
So the key issue here is what is the purpose of removing `health_check` in arch-migration step.

Reply:
> sometimes the MCP will not ready even cluster status reported Progressing=false

This is expected behavior, or some known issue, or some potential issue which is reporting a false positive result.

Reply:
> We could not make sure all things ready if we keep cucushift-ota-postupgrade just behind cucushift-upgrade-arch-migration for OCP 4.18 and later versions.

I see check_co was removed now, so it seems not a problem and `health_check` could be restored, right?

Keeping postupgrade just behind the migration is because the postupgrade will not exit even fail so it will not block next step, but `ipi-install-heterogeneous` fail will break the whole job, then post-check steps in ota cases will be blocked, which is not expected.

Reply:
The key point to the PR is add extra worker node immediately, if we put postupgrade before ipi-install-heterogeneous will break the rule.

Reply:
maybe we can update the job that make sure `to-multiarch` jobs do not enable any OTA cases, then we can move `cucushift-ota-postupgrade` before `ipi-install-heterogeneous`. 
I checked below job:
```
- as: aws-ipi-ovn-ipsec-to-multiarch-f28
  cron: 44 18 16 * *
  steps:
    cluster_profile: aws-qe
    env:
      ADDITIONAL_WORKERS: "1"
      BASE_DOMAIN: qe.devcluster.openshift.com
      COMPUTE_NODE_REPLICAS: "2"
      ENABLE_OTA_TEST: OCP-60396
    test:
    - chain: openshift-upgrade-qe-sanity-arch-migration
    workflow: cucushift-installer-rehearse-aws-ipi-ovn-ipsec
```
it do:
1. inspect docker image
2. check rollback should be reject
I think they may take some minutes

So I think we can merge all OTA cases which currently in different `to-multiarch` jobs into one.

Reply:
> The key point to the PR is add extra worker node immediately

It sounds like a dedicated checkpoint for the test only, right? Personally i think it's not a general scenario that there is not any health check and post checks between migration and extra worker adding. Even in real customer env, it's not the case.

1. There is also `check_arch` and `health_check` func after migration in `cucushift-upgrade-arch-migration`, so it's not strictly immediately either, we could regard postupgrade a similar func like them, and it bring very little delay(for example: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-stable-azure-ipi-to-multiarch-f28/1854169195600678912).

```
INFO[2024-11-06T16:30:56Z] Running step azure-ipi-to-multiarch-f28-cucushift-ota-postupgrade. 
INFO[2024-11-06T16:31:58Z] Step azure-ipi-to-multiarch-f28-cucushift-ota-postupgrade succeeded after 1m2s.
```

2. If we really could not accept any steps between migration and extra worker adding, then maybe we need consider to introduce an env for this checkpoint only(sth like add_extra_worker_immediately), once this env enabled, ignore all steps between them. Or even copying another chain for this test only.

Reply:
[Look at the test plan on https://issues.redhat.com/browse/OTA-1386 and https://github.com/openshift/enhancements/blob/master/enhancements/update/duration-of-migration-to-multi-arch.md#test-plan, this is required for 4.17+](https://github.com/openshift/release/pull/58716#discussion_r1841899026)

Reply:
Adding extra worker nodes immediately is what our customers do.

Reply:
In commit https://github.com/openshift/release/compare/f5451f4f1cab789ea7b168c62a58be23b9131e60..80f55ca44e10d2493f430af74d957830608f369b, I cleaned some steps in post-OCP-53921, it can saves a little time

Reply:
Why we need disrupt OCP-53921?

Reply:
Because:
1. OCP-53921 only enabled in `to-multiarch` jobs
2. in `cucushift-upgrade-arch-migration-commands.sh`, we already have `switch_channel()` and `check_arch()`

So we don't need to waste time to do same work twice

Reply:
I mean we are talking the better sequence for the chain, the case's checkpoint does not affect the sequence. As for the case OCP-53921, because there are some other checkpoints in the case, we don't want to check the case's test result in several places, so there is also once more arch check in the postcheck step, i think it does no harm to keep it there.

Reply:
https://github.com/openshift/release/pull/58716#discussion_r1843073690